src/main/backend.ts

```diff
@@ -150,30 +142,95 @@ export class MainBackend extends BaseBackend<ElectronOptions> implements CommonB
     const hubScope = getCurrentHub().getScope();
     if (hubScope) {
       hubScope.addScopeListener(updatedScope => {
-        const cloned = Scope.clone(updatedScope);
+        const scope = Scope.clone(updatedScope);
         // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-        (cloned as any)._eventProcessors = [];
+        (scope as any)._eventProcessors = [];
         // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-        (cloned as any)._scopeListeners = [];
+        (scope as any)._scopeListeners = [];
+
         // if we use the crashpad minidump uploader we have to set extra whenever the scope updates
         if (this._options.useCrashpadMinidumpUploader === true) {
-          getCurrentHub().captureEvent(
-            {
-              // @ts-ignore __INTERNAL_MINIDUMP is not assignable to event
-              __INTERNAL_MINIDUMP: true,
-            },
-            {
-              data: {
-                __sentry__: true,
-              },
-            },
+          forget(
+            this._getNativeUploaderEvent(scope).then(event => {
```

Comment:
This starts an asynchronous operation. Can we prevent race conditions from consecutive updates?

Reply:
Good point 🤔